### ISSUE TITLE
Error with graph in in normalizing-flows/normalizing_flows/flows.py

### ISSUE THREAD

**Issue by niharika2999** [davidnabergoj/normalizing-flows#8]
Hello! 

Thank you so much for developing this package. I was using this for my implementation of a probabilistic autoencoder and encountered an error in normalizing-flows/normalizing_flows/flows.py at line 212.  train_loss.backward() throws an error : "Trying to backward through the graph a second time (or directly access saved tensors after they have already been freed). Saved intermediate values of the graph are freed when you call .backward() or autograd.grad(). Specify retain_graph=True if you need to backward through the graph a second time or if you need to access saved tensors after calling backward". 

After going through examples of this error online I tried changing it to train_loss.backward(retain_graph = True). This did not rectify the error for me. It would be great if you could look into this error.

Thank you so much!

**Comment by davidnabergoj** [comment]
Hi, thanks for posting the issue. I'd be glad to help. Could you provide a minimal reproducible example where this error occurs? It will help narrow down the problem.

**Comment by niharika2999** [comment]
Thank you so much for your prompt response!I have posted a snippet of my code. The input to flow.fit is the latent space of my autoencoder. It is a vector with size = 100. 
<img width="830" alt="Screenshot 2024-07-03 at 12 56 02 PM" src="https://github.com/davidnabergoj/normalizing-flows/assets/87704270/b4de1cc1-65a4-4889-8b39-132553a95eb2">
<img width="1605" alt="Screenshot 2024-07-03 at 12 56 16 PM" src="https://github.com/davidnabergoj/normalizing-flows/assets/87704270/a6524076-c5b6-4e56-9a9a-5c041ee321d3">

**Comment by davidnabergoj** [comment]
Thanks for the snippet. There might be a problem with the `features` variable. For reference, I ran the following snippet on the latest version of the master branch without errors, where I replaced `features` with a random normal tensor:

```python
import torch
from normalizing_flows import Flow
from normalizing_flows.architectures import NICE

torch.manual_seed(0)
n_dim = 100
bijection = NICE(n_dim)
features = torch.randn(size=(5, n_dim))

flow = Flow(bijection)
flow.fit(features, lr=0.001, show_progress=True)
x_new = flow.sample(100)
```

Can you double check that everything is fine with the `features` variable? It might have already been used in some autograd operations prior to fitting the flow.

**Comment by niharika2999** [comment]
Hello! Thank you so much for your prompt reply. You are absolutely correct. The features variable was from the latent space of my encoder and had previous autograd operations. Just features.detach() before fitting the flow made it work. Thank you for all your help and developing this package!

**Comment by davidnabergoj** [comment]
No problem, thanks for the issue.
